OpenApi: Config quarkus.smallrye-openapi.additional-docs-directory casues error if directory does not exist

### Describe the bug

Configure `quarkus.smallrye-openapi.additional-docs-directory` to point to a directory which does not exist (e.g. `META-INF/foldernamethatdoesnotexist` .
Then run Quarkus in dev mode or try running a @QuarkusTest with `mvn clean test`.

### Expected behavior

Quarkus should ignore the missing directory and not load any additional docs.

### Actual behavior

Quarkus throws an exception:

java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor#configFiles threw an exception: java.lang.IllegalStateException: java.nio.file.NoSuchFileException: C:\dev\repo\...\target\classes\META-INF\openapi
        at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:896)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
        at java.base/java.lang.Thread.run(Thread.java:833)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.nio.file.NoSuchFileException: C:\dev\repo\...\target\classes\META-INF\openapi
        at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:85)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)


### How to Reproduce?

Set the given config property to point to a directory that does not exists and try to build/test the Quarkus project.

### Output of `uname -a` or `ver`

Windows 10

### Output of `java -version`

openjdk version "17.0.1" 2021-10-19 LTS OpenJDK Runtime Environment Microsoft-28056 (build 17.0.1+12-LTS)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.6.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.3 (ff8e977a158738155dc465c6a97ffaf31982d739)

### Additional information

_No response_

Quick note: The reproducer also fails on Debian under WSL, with the same error/exception:

...
Caused by: java.nio.file.NoSuchFileException: /mnt/c/Users/.../openapi-folder-error/target/classes/META-INF/foldernamethatdoesnotexist

--- a/extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java
+++ b/extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java
@@ -171,7 +171,8 @@ public class SmallRyeOpenApiProcessor {
             List<Path> additionalStaticDocuments = openApiConfig.additionalDocsDirectory.get();
             for (Path path : additionalStaticDocuments) {
                 // Scan all yaml and json files
-                List<String> filesInDir = getResourceFiles(path.toString(), outputTargetBuildItem.getOutputDirectory());
+                List<String> filesInDir = getResourceFiles(PathUtils.asString(path, "/"),
+                        outputTargetBuildItem.getOutputDirectory());
                 for (String possibleFile : filesInDir) {
                     watchedFiles.produce(new HotDeploymentWatchedFileBuildItem(possibleFile));
                 }
@@ -845,10 +846,11 @@ public class SmallRyeOpenApiProcessor {
             Path classes = target.resolve("classes");
             if (classes != null) {
                 Path path = classes.resolve(pathName);
-
-                return Files.list(path).map((t) -> {
-                    return pathName + "/" + t.getFileName().toString();
-                }).collect(Collectors.toList());
+                if (Files.exists(path)) {
+                    return Files.list(path).map((t) -> {
+                        return pathName + "/" + t.getFileName().toString();
+                    }).collect(Collectors.toList());
+                }
             }
         }
         return filenames;

--- a/extensions/smallrye-openapi/deployment/src/test/java/io/quarkus/smallrye/openapi/test/jaxrs/OpenApiEmptyStaticModelsTestCase.java
+++ b/extensions/smallrye-openapi/deployment/src/test/java/io/quarkus/smallrye/openapi/test/jaxrs/OpenApiEmptyStaticModelsTestCase.java
@@ -1,0 +1,34 @@
+package io.quarkus.smallrye.openapi.test.jaxrs;
+
+import java.io.IOException;
+
+import org.hamcrest.Matchers;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.restassured.RestAssured;
+
+public class OpenApiEmptyStaticModelsTestCase {
+
+    private static String directory = "META-INF/foldernamethatdoesnotexist";
+
+    @RegisterExtension
+    static QuarkusUnitTest runner = new QuarkusUnitTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClasses(OpenApiResource.class, ResourceBean.class)
+                    .addAsResource(
+                            new StringAsset("quarkus.smallrye-openapi.additional-docs-directory=" + directory),
+                            "application.properties"));
+
+    @Test
+    public void testNonExistingAdditionalDocsDirectory() throws IOException {
+
+        RestAssured.given().header("Accept", "application/json")
+                .when().get("/q/openapi")
+                .then()
+                .log().body().and()
+                .body("openapi", Matchers.startsWith("3.0."));
+    }
+}

